File: backend/src/models/models.service.ts
import { Injectable } from '@nestjs/common';
import { SupabaseService } from '../supabase/supabase.service';

@Injectable()
export class ModelsService {
  constructor(private readonly supabaseService: SupabaseService) {}

async getModels(categoryId?: string) {
  let query = this.supabaseService.client
    .from('models')
    .select('*')
    .order('id', { ascending: true });

  if (categoryId) {
    query = query.eq('category_id', categoryId); // ✅ FIX
  }

  const { data, error } = await query;

  if (error) {
    console.error(error); // 🔥 خليها
    throw new Error(`Failed to fetch models: ${error.message}`);
  }

  return data;
}

async getModelById(id: string) {
  const { data, error } = await this.supabaseService.client
    .from('models')
    .select('*')
    .eq('id', id) // ✅ FIX
    .single();

  if (error) {
    console.error('SUPABASE ERROR:', error); // 🔥 مهم
    throw new Error(error.message);
  }

  return data;
}
}

File: backend/app/services/models_service.py
from app.services.supabase_service import supabase_service


def get_models(category_id: str | None = None) -> list[dict]:
    query = supabase_service.client.table("models").select("*").order("id", desc=False)

    if category_id:
        query = query.eq("category_id", category_id)

    response = query.execute()
    return response.data or []


def get_model_by_id(model_id: str) -> dict:
    response = (
        supabase_service.client.table("models")
        .select("*")
        .eq("id", model_id)
        .single()
        .execute()
    )

    if not response.data:
        raise ValueError("Model not found")

    return response.data
